ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/serde/json/ValueSpecJsonSerdeSupplier.java

```diff
@@ -43,10 +43,18 @@
       .enable(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS)
       .setNodeFactory(JsonNodeFactory.withExactBigDecimals(true));
 
+  private static final ObjectMapper FLOAT_MAPPER = new ObjectMapper();
+
   private final boolean useSchemas;
+  private final ObjectMapper mapper;
 
-  public ValueSpecJsonSerdeSupplier(final boolean useSchemas) {
+  public ValueSpecJsonSerdeSupplier(
+      final boolean useSchemas,
+      final Map<String, Object> properties
+  ) {
     this.useSchemas = useSchemas;
+    mapper = (boolean) (properties.getOrDefault("use.big.decimal.for.floats", true))
```

Comment:
I had to update `geodistance` tests to using floating point comparisons instead of decimals since it seems like some of the floating point operations have changed precision in the different JDKs. This gives us that flexibility. The `geodistance` test used doubles (not decimals) anyway.

Reply:
Some operations changed precision but we're using them anyway?  That seems kind of crazy.  I thought most of the operations are well defined standard operations on IEEE floats.

Reply:
>  but can't you also implement that fudge factor without disabling BigDecimal

we should not be fudging comparisons with decimals, as that would explicitly defeat their purpose! I'm not sure I'm understanding your question?

Reply:
Thanks, @agavra , we just synced offline about this. Just to document what I was getting at was that it seems a bit abstract to say "I'm going to disable BigDecimal here" for the underlying reason that I happen to know that disabling BigDecimal means that the test suite will fall back to Double, and further that if it falls back to Double, it'll only compare with 6 digits of precision, when all I'm really after is the 6-digit precision.

In contrast, if we add a config that says "compute result equality +/- 0.000001", it'll be crystal clear what the test spec is doing. Then, we wouldn't need that explanatory comment in the spec, it'll be clear which tests are precise or not (regardless of what type they use), it'll be clear for future devs how to adjust their tests to account for precision errors, and we won't be inadvertantly relaxing the validations for all our other tests.

Reply:
I updated this to `use.exact.numeric.comparsion=false`, I ultimately buy your argument :)

Reply:
Thanks for hearing me out :) Sorry it took me a while to find the words for my feelings.